File: algorithms/572. Subtree of Another Tree/solution.cc
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSubtree(TreeNode* s, TreeNode* t) {
        string strs, strt;
        serialize(s, strs);
        serialize(t, strt);
        return (',' + strs).find(',' + strt) != string::npos;
    }

    void serialize(TreeNode * root, string & out) {
        if ( root ) {
            out += "," + to_string(root->val);
            serialize(root->left, out);
            serialize(root->right, out);
        } else {
            out += ",#";
        }
    }
};

// 174 / 174 test cases passed.
// Status: Accepted
// Runtime: 32 ms

File: algorithms/572. Subtree of Another Tree/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isSubtree(self, s, t):
        """
        :type s: TreeNode
        :type t: TreeNode
        :rtype: bool
        """
        def dfs(root):
            if not root:
                return "null"
            return "%s,%s,%s" % (root.val, dfs(root.left), dfs(root.right))
        return (',' + dfs(s)).find(',' + dfs(t)) != -1

# 174 / 174 test cases passed.
# Status: Accepted
# Runtime: 95 ms
